### PR TITLE
feat: Added Spark history server apis

### PR DESCRIPTION
**Pull request by silpidhua**
feat: Added Spark history server apis

## PR summary
Added Spark History server `start`, `stop` and `get` api in go sdk

**Fixes:** <! -- link to issue -->

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [ ] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?    
- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by silpidhua** [comment]
image
image
image
image

**Comment by silpidhua** [comment]
image

File: HOW_TO_TEST.md [hunk 1]
```diff
@@ -1,16 +1,26 @@
 # How to run tests
 
 ## Pre-requisites
+1. Ensure that your JAVA_HOME is set correctly
+    `export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_261.jdk/Contents/Home` 
+2. Download [SDK generation code](https://github.ibm.com/CloudEngineering/openapi-sdkgen/releases) 
+3. Clone [`GO SDK`] https://github.com/IBM/ibm-iae-go-sdk/
+4. Generate the sdk code with: 
+`openapi-sdkgen/openapi-sdkgen.sh  generate -g ibm-go -i ibmanalyticsengine-v3.json  -o /Users/git/ibm-iae-go-sdk  --genITs --genExamples --api-package github.com/IBM/ibm-iae-go-sdk`
```
Comment (ShyamalaGowri) [inline HOW_TO_TEST.md:9]: Lets add a note(hyper link) as to where this spec `ibmanalyticsengine-v3.json` can be downloaded from

File: HOW_TO_TEST.md [hunk 2]
```diff
@@ -1,16 +1,26 @@
 # How to run tests
 
 ## Pre-requisites
+1. Ensure that your JAVA_HOME is set correctly
+    `export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_261.jdk/Contents/Home`
```
Comment (ShyamalaGowri) [inline HOW_TO_TEST.md:5]: Lets add a note to set Go Path

File: HOW_TO_TEST.md (hunk 1 above)
Comment (ShyamalaGowri) [reply HOW_TO_TEST.md:9]: We also need to give some details on building the GO Module. 
The integration  tests have to run on the generated sdk and not the sdk from git. (Let have those instructions also as part of the readme)

**Comment by ShyamalaGowri** [comment]
Apart from the changes in `[ibmanalyticsengineapiv3/ibm_analytics_engine_api_v3_integration_test.go](https://github.com/IBM/ibm-iae-go-sdk/pull/18/files#diff-b8595dc7dea2ae21fa4064ac5bb1b2221a8a17f60d07d24c961ed7c9be1eede3)` all the other changes are by the sdk generation code.

ibm_analytics_engine_api_v3_integration_test.go is modified to test it with a deployed serverless instance (ys1 or prod)

File: ibmanalyticsengineapiv3/ibm_analytics_engine_api_v3_integration_test.go [hunk 3]
```diff
@@ -47,11 +47,13 @@ var _ = Describe(`IbmAnalyticsEngineApiV3 Integration Tests`, func() {
 		ibmAnalyticsEngineApiService *ibmanalyticsengineapiv3.IbmAnalyticsEngineApiV3
 		serviceURL   string
 		config       map[string]string
+		// !!! Start of custom content to be copied !!!
```
Comment (ShyamalaGowri) [inline ibmanalyticsengineapiv3/ibm_analytics_engine_api_v3_integration_test.go:50]: lets add a small note on why these changes are required

**Review by ShyamalaGowri — CHANGES_REQUESTED** [review]

File: HOW_TO_TEST.md (hunk 2 above)
Comment (silpidhua) [reply HOW_TO_TEST.md:5]: Added how to set Go Path

File: HOW_TO_TEST.md (hunk 1 above)
Comment (silpidhua) [reply HOW_TO_TEST.md:9]: Added Cloud api docs.

Comment (silpidhua) [reply HOW_TO_TEST.md:12]: Added note to run the test on generated sdk

File: ibmanalyticsengineapiv3/ibm_analytics_engine_api_v3_integration_test.go (hunk 3 above)
Comment (silpidhua) [reply ibmanalyticsengineapiv3/ibm_analytics_engine_api_v3_integration_test.go:50]: Added note on this

**Comment by silpidhua** [comment]
For the issue: https://github.ibm.com/hummingbird/tracker/issues/3191

**Comment by subinpc** [comment]
Closing as these are redundant.
